psi4/src/psi4/libmints/thc_eri.cc

```diff
@@ -259,6 +259,51 @@ SharedMatrix LS_THC_Computer::build_E_df() {
     return linalg::triplet(E_temp, Jinv, E_temp, false, false, true);
 }
 
+/// @brief Drastically reduces the size of the grid using pivoted Cholesky decomposition (Parrish et al. 2012 procedure 2)
+SharedMatrix LS_THC_Computer::prune_grid() {
+
+    outfile->Printf("    Pruning grid using pivoted Cholesky (Matthews 2020)...\n");
+
+    size_t npoints = x1_->nrow();
+    size_t nbf = x1_->ncol();
+
+    SharedMatrix S_Qq = std::make_shared<Matrix>(npoints, npoints);
+    SharedMatrix S_temp = linalg::doublet(x1_, x1_, false, true);
+
+    auto S_Qq_p = S_Qq->pointer();
+    auto Stp = S_temp->pointer();
+
+    // Parrish 2012, Equation 28
+#pragma omp parallel for
```

Comment:
```suggestion
#pragma omp parallel for collapse(2) schedule(guided)
```

Probably not necessary, but this will perform the multithreading over both loop indices simultaneously (`collapse(2)`) while dynamically adjusting batch size to prevent the scheduler from becoming the bottleneck (`guided`).